app/javascript/mastodon/features/ui/components/image_loader.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import ImmutablePureComponent from 'react-immutable-pure-component';
+import PropTypes from 'prop-types';
+
+class ImageLoader extends ImmutablePureComponent {
```

Comment:
We've been using ImmutablePureComponent throughout the codebase. I'm not 100% sure on the difference, but I like ImmutablePureComponent because it has nice settings for specifying which props/states to diff.